cuid2_histogram_test.go

```diff
@@ -0,0 +1,83 @@
+package cuid2
+
+import (
+	"fmt"
+	"math"
+	"math/rand/v2"
+	"strings"
+	"testing"
+)
+
+func TestHistogram(t *testing.T) {
+	n := 100000
+	fmt.Printf("Testing %d unique ids\n", n)
+	poolId := rand.IntN(100)
+	poolResponse := CreateIdPool(t, n, poolId)
+	ids := poolResponse.Ids
+	sampleIds := ids[:10]
+	fmt.Printf("Sample ids %v\n", sampleIds)
+	t.Run("Test collision", func(t *testing.T) {
+		CheckCollision(t, ids)
+	})
+	t.Run("Test char frequency", func(t *testing.T) {
+		testCharFrequency(t, n, ids)
+	})
+	t.Run("Test histogram", func(t *testing.T) {
+		testHistogram(t, poolResponse, n)
+	})
+
+}
+
+func testCharFrequency(t *testing.T, n int, ids []string) {
+	tolerance := 0.1
+	idLength := 23
+	totalLetters := idLength * n
+	base := 36
+	expectedBinSize := math.Ceil(float64(totalLetters) / float64(base))
+	minBinSize := math.Round(expectedBinSize * (1 - tolerance))
+	maxBinSize := math.Round(expectedBinSize * (1 + tolerance))
+
+	// Drop the first character because it will always be a letter, making
+	// the letter frequency skewed.
+	testIds := make([]string, len(ids))
+	for i, id := range ids {
+		testIds[i] = id[2:]
+	}
```

Comment:
for dropping the first letter slicing should be from 1, so it should be...

```go
for i, id := range ids {
	testIds[i] = id[1:]
}
```

Reply:
The original code had substring of 2, so I kept 2. But it make sense to have 1. Changed.